plugins/woocommerce/client/blocks/assets/js/blocks/product-template/style.scss

```diff
@@ -83,3 +83,27 @@ $min-product-width: 150px;
 .is-product-collection-layout-list .wc-block-product:not(:last-child) {
 	margin-bottom: 1.2rem;
 }
+
+/**
+ * Carousel layout
+ */
+.is-product-collection-layout-carousel {
+	display: flex;
+	flex-direction: row;
+	flex-wrap: nowrap;
+	gap: $grid-gap;
+	overflow-x: auto;
+	// TODO: Ultimately we'll hide the scrollbar as there will be arrows to navigate.
+	// For now we keep the scrollbar visible for testing purposes.
+	// scrollbar-width: none;
+	scroll-snap-type: x mandatory;
+	scroll-padding: 0 30%;
+	position: relative; // Create new stacking context
+
+	.wc-block-product {
+		scroll-snap-align: center;
+		min-width: $min-product-width;
+		flex-basis: 0;
+		flex-grow: 1;
+	}
+}
```

Comment:
I don't know if you tried, but instead of using this custom CSS, could [we try to use the `layout` property](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-supports/#layout-default) on the Product Template block?

Reply:
So I tried doing this in this PR: https://github.com/woocommerce/woocommerce/pull/59647. Basically setting up `layout` settings on switch to carousel and resetting it to empty object on switch from carousel.

But we need to remember two other layouts: grid and list and handled manually, not through `supports.layout`. And while we would like to switch to native `supports.layout` (there's been attempts and effort to do it: https://github.com/woocommerce/woocommerce/pull/46255) I somehow worry mixing these two may create more confusion or problems 🤔 Can't think of any and maybe it's just a step towards it? 😄 

Happy to hear your thoughts if you see any risks on this or you think switching to native `supports.layout` at least for Carousel is already a move in right direction?

Reply:
I don't have a lot of experience with the `supports.layout` API. From my understanding, WordPress takes care to generate the CSS. Do you see some particular issues? 


Especially if, in the long term, we want to use the `supports.layout` I think that we should move towards it, at least for the new layout.

Reply:
Sounds great! Thanks, mate! 🙇